programs/intent-transfer/src/config/state/fee_config.rs

```diff
@@ -0,0 +1,95 @@
+use anchor_lang::prelude::*;
+use anchor_spl::token::spl_token::try_ui_amount_into_amount;
+use anchor_spl::token::{transfer_checked, Mint, Token, TokenAccount, TransferChecked};
+use solana_intents::SymbolOrMint;
+
+use crate::{error::IntentTransferError, verify::verify_symbol_or_mint};
+
+pub const FEE_CONFIG_SEED: &[u8] = b"fee_config";
+
+#[account]
+#[derive(InitSpace)]
+pub struct FeeConfig {
+    pub ata_creation_fee: u64,
+    pub bridging_out_fee: u64,
+}
+
+pub struct VerifyAndCollectAccounts<'a, 'info> {
+    pub fee_source: &'a Account<'info, TokenAccount>,
+    pub fee_destination: &'a Account<'info, TokenAccount>,
+    pub fee_mint: &'a Account<'info, Mint>,
+    pub fee_metadata: &'a Option<UncheckedAccount<'info>>,
+    pub intent_transfer_setter: &'a UncheckedAccount<'info>,
+    pub token_program: &'a Program<'info, Token>,
+}
+
+impl FeeConfig {
+    pub fn verify_and_collect_ata_creation_fee(
+        &self,
+        args: VerifyAndCollectAccounts,
+        fee_amount: String,
+        fee_symbol_or_mint: SymbolOrMint,
+        signer_seeds: &[&[&[u8]]],
+    ) -> Result<()> {
+        self.verify_and_collect_fee(
+            self.ata_creation_fee,
+            args,
+            fee_amount,
+            fee_symbol_or_mint,
+            signer_seeds,
+        )
+    }
+
+    pub fn verify_and_collect_bridging_out_fee(
+        &self,
+        args: VerifyAndCollectAccounts,
+        fee_amount: String,
+        fee_symbol_or_mint: SymbolOrMint,
+        signer_seeds: &[&[&[u8]]],
+    ) -> Result<()> {
+        self.verify_and_collect_fee(
+            self.bridging_out_fee,
+            args,
+            fee_amount,
+            fee_symbol_or_mint,
+            signer_seeds,
+        )
+    }
+
+    fn verify_and_collect_fee(
```

Comment:
The abstraction here seems incorrect to me, this shouldn't be a method in the `FeeConfig` impl.  No reference to the fee config (`self`) is used anywhere.

I think a better abstraction would be to make a trait with the method `fee_amount` and implement that trait on the two account structs, and then you can implement a default `verify_and_collect_fee` in terms of that `fee_amount`